src/main/java/io/kontur/disasterninja/notifications/email/EmailMessageFormatter.java

```diff
@@ -0,0 +1,86 @@
+
+import io.kontur.disasterninja.dto.EmailDto;
+import io.kontur.disasterninja.dto.eventapi.EventApiEventDto;
+import io.kontur.disasterninja.dto.eventapi.FeedEpisode;
+import io.kontur.disasterninja.notifications.MessageFormatter;
+import io.micrometer.core.instrument.util.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+
+import static io.kontur.disasterninja.util.FormatUtil.formatNumber;
+import static org.apache.commons.lang3.text.WordUtils.capitalizeFully;
+
+@Component
+@ConditionalOnProperty(value = "notifications.enabled")
+public class EmailMessageFormatter extends MessageFormatter {
+
+    private final static Logger LOG = LoggerFactory.getLogger(EmailMessageFormatter.class);
+
+    private final static DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("MMMM d, yyyy");
+
+    @Value("${notifications.konturUrlPattern}")
+    private String konturUrlPattern;
+
+    @Value("${notifications.feed}")
+    private String feed;
+
+    public EmailDto format(EventApiEventDto event, Map<String, Object> urbanPopulationProperties,
+                            Map<String, Double> analytics) throws IOException {
+        String textTemplate = loadTemplate("notification/gg-email-template.txt");
+        String htmlTemplate = loadTemplate("notification/gg-email-template.html");
+
+        FeedEpisode lastEpisode = getLatestEpisode(event);
+
+        return new EmailDto(
+                createSubject(event, lastEpisode),
+                replacePlaceholders(textTemplate, event, lastEpisode, urbanPopulationProperties),
+                replacePlaceholders(htmlTemplate, event, lastEpisode, urbanPopulationProperties));
+    }
+
+    private String loadTemplate(String templateName) throws IOException {
+        try (InputStream inputStream = getClass().getClassLoader().getResourceAsStream(templateName)) {
+            if (inputStream == null) {
+                throw new IllegalArgumentException("Failed to find notification template: " + templateName);
+            }
+            return IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+        }
+    }
+
+    private String createSubject(EventApiEventDto event, FeedEpisode lastEpisode) {
+        StringBuilder sb = new StringBuilder();
+        sb.append(getMessageColorCode(event, lastEpisode, true));
+        sb.append(getEventStatus(event));
+        sb.append(event.getName());
+        return sb.toString();
+    }
+
+    private String replacePlaceholders(String template, EventApiEventDto event, FeedEpisode lastEpisode, Map<String, Object> urbanPopulationProperties) {
```

Comment:
What do you think about https://www.baeldung.com/spring-email-templates instead of manually implementing substitutions?

Reply:
For now it's ok, I didn't want to spend much time on it for basic version, we don't need any features that frameworks give right now. When we improve the template, then we'll for sure use some tool instead of manual replacement of values